src/mission_ui.cpp

```diff
@@ -71,9 +71,14 @@ class mission_ui_impl : public cataimgui::window
         mission_ui_tab_enum selected_tab = mission_ui_tab_enum::ACTIVE;
         mission_ui_tab_enum switch_tab = mission_ui_tab_enum::num_tabs;
 
-        size_t window_width = str_width_to_pixels( TERMX ) / 2;
-        size_t window_height = str_height_to_pixels( TERMY ) / 2;
-        size_t table_column_width = window_width / 2;
+        float viewport_width = ImGui::GetMainViewport()->Size.x;
+        float viewport_height = ImGui::GetMainViewport()->Size.y;
+
+        float minimum_window_width = str_width_to_pixels( 80 );
+        float minimum_window_height = str_height_to_pixels( 24 );
+        float window_width = std::clamp( viewport_width / 2, minimum_window_width, viewport_width );
+        float window_height = std::clamp( viewport_height / 2, minimum_window_height, viewport_height );
+        float table_column_width = window_width / 2;
```

Comment:
This is class declaraction code, it's not inside a function, so making this kind of imperative style code feels off. You are not using most of these class fields anywhere else, so there is no reason for them to be class fields at all.

You can put all of this into `draw_controls()` (which is the only place that references those).

(I would also suggest cutting down on the number of named variables here, because they might add more clutter than they give clarity, but that's up to you at this point)

(I know you're extending someone else's code, so it feels natural to write it like this. It just grew from "a small hack that we can probably ignore" to "this is kinda ugly now")

Reply:
No problems, thanks for the feedback 👍